src/plugins/github/api.js

```diff
@@ -123,6 +135,10 @@ class Commentable<T: IssueNodePayload | PullRequestNodePayload> extends Post<
 }
 
 export class Author extends GithubEntity<AuthorNodePayload> {
+  static asType(e: Entity): Author {
+    assertEntityType(e, AUTHOR_NODE_TYPE);
```

Comment:
Yep, that matched my expectation that going off the beaten path would lead to odd flow behavior. 

This is the macro library I mentioned: [sweet.js]

[sweet.js]: https://www.sweetjs.org/